### PR TITLE
document that `torch.triangular_solve` has optional out= parameter

### LINKED ISSUE #57955: triangular_solve should accept an out parameter.
## 🚀 Feature

The `torch.triangular_solve` function forces creation of a new tensor and does not allow for overriding the original `B` tensor passed in as an argument.

Adding an `out` argument should optionally give the user more control over whether or not to override the input, similar to `overwrite_b` in `scipy.linalg.solve_triangular`.

## Motivation

Needed to perform destructive triangular solve for a linalg routine but ended making copies instead.

## Alternatives

Arguments similar to `overwrite_b` are also an option.

### PR DESCRIPTION
**Pull request by dagitses**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63253

Fixes #57955

Differential Revision: [D30312134](https://our.internmc.facebook.com/intern/diff/D30312134)

### CODE REVIEW

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30312134).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30312134).

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -9739,6 +9739,7 @@ def merge_dicts(*dicts):
     unitriangular (bool, optional): whether :math:`A` is unit triangular.
         If True, the diagonal elements of :math:`A` are assumed to be
         1 and not referenced from :math:`A`. Default: ``False``.
+    out ((Tensor, Tensor), optional): the output tensors
```
Comment (mruberry) [inline torch/_torch_docs.py:9742]: To properly document the out parameter it should also appear in the signature.

"out" is a Keyword arg, which we typically separate from the "Args"

Comment (dagitses) [reply torch/_torch_docs.py:9744]: thanks, i should have looked at some other examples first!

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30312134).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30312134).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30312134).

**Review by mruberry — APPROVED** [review]
Just two minor suggestions inline for your review

Pre-approving for code velocity

**Pull request closed without merging**